ffi/diplomat/src/provider.rs

```diff
@@ -252,30 +308,34 @@ pub mod ffi {
         )]
         pub fn enable_locale_fallback(&mut self) -> DiplomatResult<(), ICU4XError> {
             match core::mem::take(&mut self.0) {
-                ICU4XDataProviderInner::Empty => Err(icu_provider::DataErrorKind::MissingDataKey
-                    .into_error()
-                    .into()),
+                ICU4XDataProviderInner::Destroyed => Err(icu_provider::DataError::custom(
+                    "This provider has been destroyed",
+                )),
+                ICU4XDataProviderInner::Empty => {
+                    Ok(ICU4XDataProvider(ICU4XDataProviderInner::Empty))
+                }
+                #[cfg(not(feature = "any_provider"))]
+                ICU4XDataProviderInner::Baked => panic!(
+                    "Locale fallback for baked providers requires the 'any_provider' feature"
+                ),
```

Comment:
Thought/Suggestion: You can do this without `any_provider` by adding another variant to the enum, `BakedFallback` or something like that. I think it's a common enough case to warrant that.

Nit: I think `panic!` might be too strong for this case

Reply:
Ah I didn't try exactly that, I tried `Baked(Option<LocaleFallbacker>)`. I guess having a separate variant will still work with DCE, but we get exponential blowup of forking.

Reply:
We need to teach diplomat about cfg, then this whole method can be `cfg(any(feature = "any_provider", feature = "buffer_provider"))`

Reply:
For now we do cfgs inside the methods

Reply:
I tried again with an extra `BakedFallback(Vec<LocaleFallbacker>)` option, but the compiler cannot figure out that a variant is never used, so it always pulls in the fallback code (which has to be in `load`).

Reply:
the variant should also be cfg'd off, yes?

Reply:
That would require a separate `baked_fallback` feature. I'm trying to reduce feature complexity these days...